lib/abci/handlers/query/identitiesByPublicKeyHashesQueryHandlerFactory.js

```diff
@@ -77,17 +80,34 @@ function identitiesByPublicKeyHashesQueryHandlerFactory(
 
     if (request.prove) {
       const proof = response.getProof();
+      const storeTreeProofs = new StoreTreeProofs();
 
       const {
         rootTreeProof,
         storeTreeProof,
       } = previousRootTree.getFullProof(
         previousIdentitiesStoreRootTreeLeaf,
-        identityIds.filter(Boolean).map((identityId) => identityId.toBuffer()),
+        identityIds.map((identityId) => {
+          if (identityId) {
+            return identityId.toBuffer();
+          }
+
+          return null;
+        }),
+      );
+
+      const {
+        storeTreeProof: publicKeyStoreTreeProof,
+      } = previousRootTree.getFullProof(
+        previousPublicKeyToIdentityIdStoreRootTreeLeaf,
+        publicKeyHashes,
       );
 
+      storeTreeProofs.setIdentitiesProof(storeTreeProof);
+      storeTreeProofs.setPublicKeyHashesToIdentityIdsProof(publicKeyStoreTreeProof);
```

Comment:
We don't need to pass the whole proofs actually. We can pass `PublicKeyHashesToIdentityIds` only for non-existing proof and `IdentitiesProof` only for existing @antouhou